### PR TITLE
[Bug fix] No Exp Gained after Switching in a Pokemon as the Opponent Faints

### PR DESCRIPTION
**Pull request by PrabbyDD**
Fix for #4667 

## What are the changes the user will see?
Will now properly give xp to all participants for this specific case

## Why am I making these changes?
bug fix

## What are the changes from a developer perspective?
Added in faint-phase a loop that adds the current pokemon to the participants list, which also happens in TurnInitPhase. 
If the battle ends as a pokemon is being switched in (such as with a poisoned enemy), then the next turnInitPhase doesnt happen and the current pokemon is not added. This fixes that. 

### Screenshots/Videos

https://github.com/user-attachments/assets/cd3d4688-43db-4311-9519-bdf3712e7a5d

## How to test the changes?
Override files, browser debugger

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] If I have text, did I make it translatable and add a key in the English locale file(s)?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/phases/faint-phase.ts [hunk 1]
```diff
@@ -65,6 +65,15 @@ export class FaintPhase extends PokemonPhase {
       }
     }
 
+    /** In case the current pokemon was just switched in, make sure it is counted as participating in the combat */
+    this.scene.getField().forEach((pokemon, i) => {
```
Comment (DayKev) [inline src/phases/faint-phase.ts:69]: You could probably use `getPlayerField()` and skip the `isPlayer()` check below.

Comment (PrabbyDD) [reply src/phases/faint-phase.ts:69]: I just did what was done in turn-init-phase. Should I still change it?

Comment (DayKev) [reply src/phases/faint-phase.ts:69]: `TurnInitPhase` also does things to the enemy's pokemon, but here you're only acting on the player's pokemon, so you can just use `getPlayerField()`.

Comment (PrabbyDD) [reply src/phases/faint-phase.ts]: I think it should be fixed now.

File: src/phases/faint-phase.ts [hunk 2]
```diff
@@ -65,6 +65,15 @@ export class FaintPhase extends PokemonPhase {
       }
     }
 
+    /** In case the current pokemon was just switched in, make sure it is counted as participating in the combat */
+    this.scene.getPlayerField().forEach((pokemon, i) => {
+      if (pokemon?.isActive()) {
```
Comment (innerthunder) [inline src/phases/faint-phase.ts:70]:
```suggestion
      if (pokemon?.isActive(true)) {
```

**Review by innerthunder — APPROVED** [review]

**Pull request merged**